clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json

```diff
@@ -19,212 +20,148 @@
-          }
+          ]
         }
       ]
     },
-    "asmLineStart": {
+    "operandsAndRemarks": {
+      "begin": "(?=\\S)(?<=^.{4,71})",
+      "end": "((?<=^.{71}\\s.*)|(?<=^.{0,71}))$",
+      "applyEndPatternLast": 0,
       "patterns": [
         {
-          "match": "^([^*]\\S*)?\\s+(.*)$",
+          "include": "#invalidMultilineBlock"
+        },
+        {
+          "match": "(^\\s{15})",
+          "comment": "Just to get rid of the spaces"
+        },
+        {
+          "include": "#sequenceSymbolOperandField"
+        },
+        {
+          "include": "#asmArguments"
+        },
+        {
+          "match": "(?<=,)\\s(.*)(?<=^.{0,71})",
+          "comment": "Operands will continue on next line",
           "captures": {
             "1": {
-              "name": "support.class.hlasm"
-            },
-            "2": {
               "patterns": [
                 {
-                  "include": "#asmInstruction"
+                  "include": "#remark"
                 }
               ]
             }
           }
+        },
+        {
+          "begin": "(?<!,)(?=\\s(?<=^.{0,71}))",
+          "end": "((?<=^.{71}\\s.*)|(?<=^.{0,71}))$",
+          "applyEndPatternLast": 1,
+          "comment": "Operands ended on this line",
+          "patterns": [
+            {
+              "include": "#remarksOrInvalidLine"
+            }
+          ]
+        },
+        {
+          "include": "#lineEnd"
         }
       ]
     },
-    "asmInstruction": {
+    "remarksOrInvalidLine": {
       "patterns": [
         {
-          "match": "(\\S+)\\s*(.*)",
-          "captures": {
-            "1": {
-              "name": "entity.name.function.hlasm"
-            },
-            "2": {
-              "patterns": [
-                {
-                  "include": "#asmArguments"
-                }
-              ]
-            }
-          }
+          "include": "#invalidMultilineBlock"
+        },
+        {
+          "include": "#remark"
+        },
+        {
+          "include": "#lineEnd"
         }
       ]
     },
-    "asmArgumentLine": {
+    "invalidMultilineBlock": {
+      "begin": "(?=^\\s*\\S(?<=^.{0,15}))|(?=^.{80}\\s*\\S)",
+      "end": "((?<=^.{71}\\s.*)|(?<=^.{0,71}))$",
+      "applyEndPatternLast": 1,
       "patterns": [
         {
-          "match": "^\\s+(\\S+=.*)$",
-          "captures": {
-            "1": {
-              "patterns": [
-                {
-                  "include": "#asmArguments"
-                }
-              ]
-            }
-          }
+          "match": "^.*",
+          "name": "invalid.illegal"
         }
       ]
     },
     "asmArguments": {
       "patterns": [
         {
-          "include": "#stringType"
+          "include": "#quoteType"
         },
         {
-          "include": "#quoteType"
+          "include": "#openEndedArgs"
         },
         {
-          "match": "(\\S*)(.*)",
+          "match": "([^\\s=(),'*/+-]+)(?<=^.{0,71})",
```

Comment:
What is the significance of `s` here? `\\s` in the `[^]` pattern probably does not do what was intended

Reply:
In this case it should force the TextMate grammar to iterate over arguments from `operandsAndRemarks` and from `parenthesis` one by one without consuming any separating spaces, commas, etc.